tasks/rsync-deploy.yml

```diff
@@ -1,19 +1,24 @@
 ---
+
+# Migration check from symlink deployment
+- stat:
+    path: "{{ ansistrano_deploy_to }}/{{ ansistrano_current_dir }}"
+  register: stat_current_dir
+
+- name: ANSISTRANO | Remove current folder if it's a symlink
+  file:
+    state: absent
+    path: "{{ ansistrano_deploy_to }}/{{ ansistrano_current_dir }}"
+  when: stat_current_dir.stat.islnk is defined and stat_current_dir.stat.islnk
+
 # Perform rsync deployment
 - name: ANSISTRANO | Ensure current folder is a directory
   file:
     state: directory
     path: "{{ ansistrano_deploy_to }}/{{ ansistrano_current_dir }}"
 
 - name: ANSISTRANO | Sync release to new current path
-  synchronize:
-    src: "{{ ansistrano_release_path.stdout }}/"
-    dest: "{{ ansistrano_deploy_to }}/{{ ansistrano_current_dir }}/"
-    set_remote_user: "{{ ansistrano_rsync_set_remote_user }}"
-    recursive: yes
-    delete: yes
-    archive: yes
-  delegate_to: "{{ inventory_hostname }}"
+  command: rsync -a -F --delete-after "{{ ansistrano_release_path.stdout }}/" "{{ ansistrano_deploy_to }}/{{ ansistrano_current_dir }}/"
```

Comment:
But you have to deal with `.ssh/known_hosts` and outgoing connections and maybe DNS issues when using rsync via ssh that's why i've replaced it with a plain rsync. It's also much faster here then running via SSH.

The rsync strategy is not the best method for deployment but a symlink is not always possible (eg. shared hosts) so my goal was to keep the trouble and downtime at a minimum. For some beginners that can be really an issue to setup a working environment.
Also there are companies that doesn't allow these connections because of security issues and they don't want to change the policy.

Eg. if you have a two digit number of servers with multiple accounts then you have to setup a valid `.ssh/known_hosts` for each account on each server for automatic deployment just because a loopback connection of one single task which can be run locally.